class 4

```diff
@@ -0,0 +1,227 @@
+/*console.log('welcome to class 4');
+
+
+console.log('first');
+console.log('second');
+setTimeout(() => {                                   // asynchronous function 
+    console.log('third');
+
+    
+}, 0);
+
+console.log('fourth');                                // out -> first second fourth third  sequentially  hoi nai 
+
+*/
+
+
+
+//----------------------------------------------------------------
+//                     Call Stack
+//console.log('first');
+//console.log('second');
+/*setTimeout(() => {                                   // asynchronous function -> call back queue te pathabe 
+
+    console.log('third');                            
+
+    
+}, 0);*/
+
+//console.log('fourth');  
+
+//console.log('third');                         // call back queue call stack e pathalo
+
+//                        |
+//                        |
+
+//                    event loop
+                        
+//                        |
+//                        |
+//-----------------------------------------------------------------
+//                    call back queue
+
+/*setTimeout(() => {                                   // asynchronous function 
+    console.log('third');
+
+    
+}, 0);*/
+
+//--------------------------------------------------------
+//                    output
+//first  
+//second  
+//fourth
+//third
+
+
+/*
+setTimeout(() => {                                   // asynchronous function 
+    console.log('hello js');
+},3000);                                    //3 se so pore
+
+setTimeout(() => {                                   
+    console.log('hello jsc');
+},1000);                                   //1sec so age jabe
+*/
+
+
+
+
+
+//const mytdo = fetch('https://jsonplaceholder.typicode.com/todos/1')             
+//console.log('mytdo::::',mytdo);    
+     
+
+
+
+
+/*   ------------------------------------------------------------------
+
+//----- --------promise creation phase  creating
+
+const myPromise = new Promise((resolve, reject) => {
+    const num = Math.ceil(Math.random()*100);
+    if (num % 2 == 0) {
+        resolve({
+            name : 'chadni',
+            email: 'jahanichadni18@gmail.com',
+            location: "dhaka",
+        });
+
+    }else{
```

Comment:
Just a simple suggestion here, try to keep space before and after `else` while you are writing `else` block this way, i.e. `} else {`